frontend/src/pages/ActivityPage.jsx

```diff
@@ -16,191 +29,207 @@ const Activity = () => {
     duration: null,
     activity: "",
   });
+  const [dataFile, setDataFile] = useState({});
 
-  const handleSubmit = () => {
-    const { age, weight, height, duration, activity } = userData;
+  // const handleSubmit = () => {
+  //   const { age, weight, height, duration, activity } = userData;
     
-    // Close the modal
-    setModalVisible(false);
-  };
+  //   // Close the modal
+  //   setModalVisible(false);
+  // };
 
-  const { met } = SampleData.SampleData;
+  // fetch the device data
+  useEffect(() => {
+    const fetchData = async () => {
+      try {
+        // service and characteristic UUIDs
+        const serviceCharacteristics = [
+          // acceleration
+          {
+            service_uuid: '00000300-1212-efde-1523-785feabcd123',
+            characteristic_uuid: '00000302-1212-efde-1523-785feabcd123'
+          },
+        ];
+
+        const response = await axios.post(`http://${ipAddress}:8000/data`, serviceCharacteristics);
+        const data = await response.data;
+        setDataFile(data);
+      } catch (error) {
+        console.error('Fetch data failed:', error);
+      }
+    };
+
+    fetchData();
+
+    const interval = setInterval(fetchData, 1000);
```

Comment:
This seems to be for real-time visualization of the data transmitted by the watch, am I correct? If so, it'd be great if `useEffect` checks if a device is connected before `setInterval` so it doesn't print error messages in dev mode

Reply:
So temporarily it will use real time data. Eventually it will pull from the data stored in SQL Lite produce averages from stored data to then display to the user. It is similar to how smartphones will display a users activity level throughout the day. But since we haven't set that up yet I just have it in real time for now so I don't think we don't have to be too worried about this until we connect the database to it.